docs/usage/test-file-structure.mdx

```diff
@@ -0,0 +1,215 @@
+- In output, when the block information is written to screen.
+:::
+
+## More complex file with test cases
+
+The file above can take advantage of `TestCases` (which are also aliased `-ForEach`) to repeat the `It` block once for every item in the provided array. This makes the `It` code repeat less, and makes it very easy to add new examples.
+
+```powershell
+# file Get-Emoji.Tests.ps1
+
+BeforeAll {
+    . $PSCommandPath.Replace('.Tests.ps1','.ps1')
+}
+
+Describe "Get-Emoji" {
+    Context "Lookup by whole name" {
+        It "Returns <expected> (<name>)" -TestCases @(
+            @{ Name = "cactus"; Expected = '🌵'}
+            @{ Name = "giraffe"; Expected = '🦒'}
+        ) {
+            Get-Emoji -Name cactus | Should -Be $expected
+        }
+    }
+
+    Context "Lookup by wildcard" {
+        Context "by prefix" {
+            BeforeAll { 
+                $emojis = Get-Emoji -Name pen*
+            }
+
+            It "Returns <expected> (<name>)" -TestCases @(
+                @{ Name = "pencil"; Expected = '✏️'}
+                @{ Name = "penguin"; Expected = '🐧'}
+                @{ Name = "pensive"; Expected = '😔'}
+            ) {
+                $emojis | Should -Contain $expected
+            }
+        }
+
+        Context "by contains" {
+            BeforeAll { 
+                $emojis = Get-Emoji -Name *happy*
+            }
+
+            It "Returns <expected> (<name>)" -TestCases @(
+                @{ Name = "slightly smiling face"; Expected = '🙂'}
+                @{ Name = "beaming face with smiling eyes"; Expected = '😁'}
+                @{ Name = "smiling face with smiling eyes"; Expected = '😊'}
+            ) {
+                $emojis | Should -Contain $expected
+            }
+        }
+    }
+}
+```
+
+See [Test cases](test-cases).
+
+## Even more complex file
+
+With the example above we are still far away from a really complex test file. And that is good. The simpler you are able to keep your tests the better. The more static your tests are, the easier it is to feel confident that everything works.
+
+You for example might notice that "by prefix" and "by contains" are very similar, and you would be able to generate them by re-using the same `Context` by just providing a different value to `-Name`. While this is technically possible using the techniques below, I would not do it. The level of complexity shown in "More complex file with test cases" is about right for 90% of tests I write.
+
+Take the examples below as a list of things you might need when you can't avoid them. Not as a list of things that you need to use in all of your tests to get "Advanced Pester user" achievement unlocked. Less is more.
+
+### Test script parameters 
+
+Passing parameters using the PowerShell `param()` block to provide external data to the `.Tests.ps1` file: 
+
+```powershell 
+param (
+    $File
+)
+
+Describe "File <file> is not empty" {
+    # ...
+}
+```
+See [Generating tests](generating-tests).
```

Comment:
Should this have referenced the non-existing `script-parameters` page that's commented out in `sidebar.js`?

Reply:
I've put it in the data driven tests, which were originally meant to be "generating tests"